GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs

```diff
@@ -93,6 +86,11 @@ public override void Update(float diff)
 
         public virtual bool AIMove()
         {
+            if (IsLaneMinion)
+            {
+                (this as ILaneMinion).WalkToDestination();
```

Comment:
This is not quite the same as the previous version. The previous version invokes the code below and then conditioning the if here based on that. What is the reason for this change?

As a general principle, if you need to use down-casting like here, then doing virtual functions is better (the old solution).

Reply:
Forgot about the virtual, and while I'm at it I'll just re-add override AIMove since I can achieve the same thing with it. My intention is to separate the Minion AI functions from the LaneMinion AI functions; Minion.ScanForTargets/KeepFocusingTarget shouldn't be used for LaneMinions.